.github/workflows/drivers.yml

```
@@ -1019,40 +1019,70 @@ jobs:
-      needs.files-changed.outputs.backend_all == 'false')
-    strategy:
-      matrix:
-        job:
-          -  "Test Postgres 12.x (App DB Tests)"
-          -  "Test Postgres 12.x (Driver Tests)"
-          -  "Test Postgres Latest (App DB Tests)"
-          -  "Test Postgres Latest (Driver Tests)"
-          -  "be-tests-mariadb-10-2-ee (0)"
-          -  "be-tests-mariadb-10-2-ee (1)"
-          -  "be-tests-mariadb-10-2-ee (2)"
-          -  "be-tests-mariadb-10-2-ee (3)"
-          -  "be-tests-mariadb-latest-ee (0)"
-          -  "be-tests-mariadb-latest-ee (1)"
-          -  "be-tests-mariadb-latest-ee (2)"
-          -  "be-tests-mariadb-latest-ee (3)"
-          -  "be-tests-mysql-8-0-ee (0)"
-          -  "be-tests-mysql-8-0-ee (1)"
-          -  "be-tests-mysql-8-0-ee (2)"
-          -  "be-tests-mysql-8-0-ee (3)"
-          -  "be-tests-mysql-latest-ee (0)"
-          -  "be-tests-mysql-latest-ee (1)"
-          -  "be-tests-mysql-latest-ee (2)"
-          -  "be-tests-mysql-latest-ee (3)"
-    name: ${{ matrix.job }}
-    runs-on: ubuntu-22.04
+  drivers-tests-result:
+    needs:
+      # if you add a driver job, you must add it to this list for it to be a required check
+      - be-tests-athena-ee
+      - be-tests-bigquery-cloud-sdk-ee
+      - be-tests-druid-ee
+      - be-tests-databricks-ee
+      - be-tests-druid-jdbc-ee
+      - be-tests-mysql-mariadb
+      - be-tests-mongo-4-4-ee
+      - be-tests-mongo-4-4-ssl-ee
+      - be-tests-mongo-5-0-ee
+      - be-tests-mongo-5-0-ssl-ee
+      - be-tests-mongo-latest-ee
+      - be-tests-mongo-sharded-cluster-ee
+      - be-tests-oracle-18-4-ee
+      - be-tests-oracle-21-3-ee
+      - be-tests-postgres-ee
+      - be-tests-presto-jdbc-ee
+      - be-tests-redshift-ee
+      - be-tests-snowflake-ee
+      - be-tests-sparksql-ee
+      - be-tests-sqlite-ee
+      - be-tests-sqlserver-2017-ee
+      - be-tests-sqlserver-2022-ee
+      - be-tests-vertica-ee
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    name: drivers-tests-result
+    if: always() && !cancelled()
     steps:
-      - run: >-
-          echo 'Skipping job due to conditional filtering'
+      - name: Check job status
+        uses: actions/github-script@v7
+        env:
+          needs: ${{ toJson(needs) }}
+        with:
+          script: | # js
+            const needs = JSON.parse(process.env.needs);
+
+            const jobs = Object.entries(needs).map(
+              ([jobName, jobValues]) => ({
+                name: jobName,
+                result: jobValues.result
+              }));
+
+            // are all jobs skipped?
+            if (jobs.every(job => job.result === 'skipped')) {
+              console.log('All jobs are skipped');
+              process.exit(0);
+            }
+
+            // are all jobs successful?
+            if (jobs.every(job => job.result === 'success')) {
+              console.log('All jobs are successful');
+              process.exit(0);
+            }
```

```suggestion
            // are all jobs skipped or successful?
            if (jobs.every(job => (job.result === 'skipped' || job.result === 'success'))) {
              console.log('All jobs are skipped or successful');
              process.exit(0);
            }
```

I think it's possible we might have some situation in the future where we skip some driver tests but not all of them. Like if I only touch BigQuery code then we could potentially only run BigQuery tests and skip everything else